include/GeographyPoint.hpp

```diff
@@ -0,0 +1,89 @@
+/* This file is part of VoltDB.
+ * Copyright (C) 2008-2016 VoltDB Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef INCLUDE_GEOGRAPHYPOINT_H_
+#define INCLUDE_GEOGRAPHYPOINT_H_
+
+namespace voltdb {
+
+class ByteBuffer;
+/**
+ * Represent a geography point. There are no virtual functions here,
+ * so there is no destructor at all.  The default one is good enough
+ * for us.
+ */
+struct GeographyPoint {
+    /**
+     * The default constructor makes a null geography.
+     */
+    GeographyPoint();
+    GeographyPoint(double aLongitude, double aLatitude)
+        : m_longitude(aLongitude),
+          m_latitude(aLatitude) {}
+
+    double getLongitude() const {
+        return m_longitude;
+    }
+
+    double getLatitude() const {
+        return m_latitude;
+    }
+
+
+    std::string toString() const;
+
+    bool isNull() const {
+        return m_longitude == 360.0 && m_latitude == 360.0;
+    }
+    
+    bool operator==(const GeographyPoint &aOther) const {
+        return getLongitude() == aOther.getLongitude() && getLatitude() == aOther.getLatitude();
```

Comment:
This will fall into the trap we hit in both Java and the EE, where points at the poles with different longitudes will not be equal, etc.